### DISCUSSION TITLE
Improve UI for deeply nested Interfaces

### DISCUSSION THREAD

**Discussion by lukas-schaetzle** [directus/directus#4220] [Feature Requests]
Hi, I really like Directus' UI, but one thing where it falls a bit short in my opinion is the handling of deeply nested relational items (e.g. many-2-any contains item with another many-to-any). In version 9, the drawers currently just overlap each other and apart from the darker background there's no indicator that multiple drawers are open. Also you can't close all drawers at once. Therefore, I have a few minor suggestions:

- a new back/up button takes the role of the current X button (only closing the top drawer)
- the X button closes all open drawers
- something like an interactive breadcrumbs bar show the current nesting

Here's a quick mockup of it, though the position of the breadcrumbs is maybe not ideal:
![Screenshot_20210222_165032](https://user-images.githubusercontent.com/45203603/108733179-8bd6fd00-752e-11eb-828b-773dc85006ee.png)

**Comment by rijkvanzanten** [comment]
> a new back/up button takes the role of the current X button (only closing the top drawer)
> the X button closes all open modals

I'm a little worried about the potential loss of data here, you could have unsaved changes in the drawer for "CollectionA" that would be discarded in the same flow 🤔

> something like an interactive breadcrumbs bar show the current nesting

Sounds like a good idea to me 🙂 We'll have to figure out what it says though, is it just the collection name, or would it show the field "path" as well (eg `Articles -> Categories (categories) -> User (directus_users)`)

---

@benhaynes regarding

> the drawers currently just overlap each other and apart from the darker background there's no indicator that multiple drawers are open.

We could try to do some sort of stacked approach where it's a little more obvious that you're "in too deep", maybe something like:

<img width="1671" alt="CleanShot 2021-02-22 at 11 02 13@2x" src="https://user-images.githubusercontent.com/9141017/108734635-a8a70c80-74fd-11eb-87a6-f605a61f3027.png">

**Reply by lukas-schaetzle** [reply]
I like the stacked drawer. Makes it very clear. Regarding the breadcrumbs name I think it makes sense to also include the field name, good point.

> 'm a little worried about the potential loss of data here, you could have unsaved changes in the drawer for "CollectionA" that would be discarded in the same flow

Couldn't we just add a warning modal dialog if there are unsaved changes?

**Reply by rijkvanzanten** [reply]
Yeah, but then we'd have to start describing where and what you did, like "You have unsaved changes in collectionA, which is 2 drawers away from this one. The changes where: A in field X, B in Field Y, etc".. Once you go that route, I think it'd be clearer to have them looking at the form with the changes they're about to undo.

I know it's a nice convenience to be able to just "let me out!!" close everything at once, but I'm wondering if it's worth the confusion in data-loss 🤔

**Reply by lukas-schaetzle** [reply]
I would not describe all pending changes but only show something like this: "There are data changes in one or multiple collection items! Are you really sure to close _all_ related items?"
But I see your point and can understand if you rate it as "too risky".

**Reply by benhaynes** [reply]
I think that "global" warning makes sense. A user should generally know if they have just made a LOT of parent/nested edits... or if they're just browsing. I don't mind the idea of a **full** "exit" button, but since the "back" buttons are stacked, you can also just click that button a few times.

I think the bigger issue to solve here is how we visually represent nested drawers. Ironically, this card design Rijk mocked is pretty close to how it was done back in v4-v5! I don't hate it, especially since we don't lose any real-estate for the "current" drawer. The only thing I would tweak is having the "under" drawers be a bit more subdued with opacity, not just the "smoke" layer.

**Reply by benhaynes** [reply]
As for the breadcrumb, we'll need to experiment with what fits in the header, but I agree that it'd be nice to see. If it gets too long we could collapse some of the middle crumbs. 

Another idea (which might look horrible) is rotating the titles 90° and adding them to the sliver of exposed underneath drawers. Not sure about this though.

Lastly, previous versions of Directus actually just used a number to represent how "deep" you were. While this isn't super helpful info, just throwing it out there.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
